File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/RegisterLocationUseCase.java
package com.dozycoffee.wms.warehouse.application.port.in;

import com.dozycoffee.wms.warehouse.application.port.in.command.RegisterLocationCommand;
import com.dozycoffee.wms.warehouse.application.port.in.result.LocationResult;
import reactor.core.publisher.Mono;

public interface RegisterLocationUseCase {

    Mono<LocationResult> register(RegisterLocationCommand command);
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/RegisterLocationUseCase.kt
package com.dozycoffee.wms.warehouse.application.port.`in`

import com.dozycoffee.wms.warehouse.application.port.`in`.command.RegisterLocationCommand
import com.dozycoffee.wms.warehouse.application.port.`in`.result.LocationResult
import reactor.core.publisher.Mono

interface RegisterLocationUseCase {
    fun register(command: RegisterLocationCommand): Mono<LocationResult>
}
